polaris-core/src/main/java/org/apache/polaris/core/persistence/BaseMetaStoreManager.java

```diff
@@ -45,4 +56,179 @@ public static PolarisStorageConfigurationInfo extractStorageConfiguration(
     return PolarisStorageConfigurationInfo.deserialize(
         callCtx.getDiagServices(), storageConfigInfoStr);
   }
+
+  /**
+   * Given the internal property as a map of key/value pairs, serialize it to a String
+   *
+   * @param properties a map of key/value pairs
+   * @return a String, the JSON representation of the map
+   */
+  public String serializeProperties(PolarisCallContext callCtx, Map<String, String> properties) {
+
+    String jsonString = null;
+    try {
+      // Deserialize the JSON string to a Map<String, String>
+      jsonString = MAPPER.writeValueAsString(properties);
+    } catch (JsonProcessingException ex) {
+      callCtx.getDiagServices().fail("got_json_processing_exception", "ex={}", ex);
+    }
+
+    return jsonString;
+  }
+
+  /**
+   * Given the serialized properties, deserialize those to a {@code Map<String, String>}
+   *
+   * @param properties a JSON string representing the set of properties
+   * @return a Map of string
+   */
+  public Map<String, String> deserializeProperties(PolarisCallContext callCtx, String properties) {
+
+    Map<String, String> retProperties = null;
```

Comment:
nit: this `null` appears to be returned on parsing exceptions. This looks odd to me... Whether `getDiagServices().fail()` throws is very non-intuitive.

Reply:
Yeah that's a good point; for now these are moved from `PolarisMetaStoreManagerImpl.java` into this shared code, so I'm trying to just preserve the old (maybe buggy) behavior in this PR. I think we have some redundant serialize/deserialize helpers in various places so maybe we can consolidate in one place in a followup.